equinox/nn/_kv_cache.py

```diff
@@ -0,0 +1,50 @@
+import jax.lax as lax
+import jax.numpy as jnp
+from jaxtyping import Array, Float, Int
+
+from .._misc import default_int_dtype
+from .._module import field, Module
+from ._stateful import State, StateIndex
+
+
+class StandardKVCache(Module):
+    key_shape: tuple[int, int, int] = field(static=True)
+    value_shape: tuple[int, int, int] = field(static=True)
+
+    autoregressive_index: StateIndex
+
+    def __init__(
+        self,
+        key_shape: tuple[int, int, int],
+        value_shape: tuple[int, int, int],
```

Comment:
Can we make the names passed in here match the names we also pass into `eqx.nn.MultiheadAttention`?

I also note that there's an implicit assumption that the first value of `key_shape` and `value_shape` are the same. Better to pull those out as a separate shared `state_length` argument I think.